blob/src/main/java/com/bazaarvoice/emodb/blob/core/DefaultBlobStore.java

```diff
@@ -349,7 +406,25 @@ public void delete(String tableName, String blobId) {
 
         Table table = _tableDao.get(tableName);
 
-        _storageProvider.deleteObject(table, blobId, null);
+        StorageSummary storageSummary = _metadataProvider.readMetadata(table, blobId);
+
+        if (storageSummary == null) {
+            throw new BlobNotFoundException(blobId);
+        }
+        _metadataProvider.deleteMetadata(table, blobId);
+        try {
+            _storageProvider.deleteObject(table, blobId);
+        } catch (Throwable t) {
+            LOGGER.error("Failed to delete blob for table: {}, blobId: {}, attempt to revert metadata deletion. Exception: {}", tableName, blobId, t.getMessage());
+            try {
+                _metadataProvider.writeMetadata(table, blobId, storageSummary);
+            } catch (Exception e1) {
+                LOGGER.error("Failed to revert metadata deletion for table: {}, blobId: {}. Inconsistency between blob and metadata storages. Exception: {}", tableName, blobId, e1.getMessage());
+                _metadataInconsistencyMeter.mark();
```

Comment:
In this case the meter comment is valid. 
Suggestion - many be have 2 different Meters for two opposite inconsistencies to Track.

Reply:
yep, will do